src/js/videojs-dash.js

```diff
@@ -115,9 +115,29 @@ class Html5DashJS {
     if (this.player.dash) {
       delete this.player.dash;
     }
+
+    Html5DashJS.beforeInitializeHooks_.length = 0;
```

Comment:
I don't think we want to reset the hooks list on dispose. The hooks are only registered once and the source handler will be disposed between sources.